src/nvim/window.c

```diff
@@ -7614,19 +7613,50 @@ void win_get_tabwin(handle_T id, int *tabnr, int *winnr)
 
 void win_ui_flush(bool validate)
 {
+  kvec_t(win_T *) stack = KV_INITIAL_VALUE;
+
   FOR_ALL_TAB_WINDOWS(tp, wp) {
     if (wp->w_pos_changed && wp->w_grid_alloc.chars != NULL) {
       if (tp == curtab) {
+        // When a floating window is anchored to another window,
+        // update the position of its anchored window first.
+        while (true) {
+          wp->w_pos_changed = false;  // avoid looping forever in case of cycle
+          if (!wp->w_floating) {
+            break;
+          }
+          FloatConfig c = wp->w_float_config;
+          if (c.external || c.relative != kFloatRelativeWindow) {
+            break;
+          }
+          Error dummy = ERROR_INIT;
+          win_T *win = find_window_by_handle(c.window, &dummy);
+          if (win == NULL) {
+            break;
+          }
+          if (win->w_pos_changed && win->w_grid_alloc.chars != NULL) {
+            kv_push(stack, wp);
+            wp = win;
+            continue;
+          }
+          break;
+        }
         ui_ext_win_position(wp, validate);
```

Comment:
`wp` here will be whatever was last assigned on line 7639, which was also added to the stack, right?
Is this line redundant?